sdk/search/azure-search-documents/src/test/java/com/azure/search/documents/FacetAggregationTests.java

```diff
@@ -0,0 +1,336 @@
+        SearchOptions defaultThreshold = new SearchOptions().setFacets("Category, metric : cardinality");
+
+        SearchOptions maxThreshold
+            = new SearchOptions().setFacets("Category, metric : cardinality, precisionThreshold: 40000");
+
+        SearchPagedIterable defaultResults
+            = getSearchClientBuilder(HOTEL_INDEX_NAME, true).buildClient().search("*", defaultThreshold, null);
+        SearchPagedIterable maxResults
+            = getSearchClientBuilder(HOTEL_INDEX_NAME, true).buildClient().search("*", maxThreshold, null);
+
+        assertNotNull(defaultResults.getFacets().get("Category"));
+        assertNotNull(maxResults.getFacets().get("Category"));
+
+        boolean defaultHasCardinality
+            = defaultResults.getFacets().get("Category").stream().anyMatch(f -> f.getCardinality() != null);
+        boolean maxHasCardinality
+            = maxResults.getFacets().get("Category").stream().anyMatch(f -> f.getCardinality() != null);
+
+        assertTrue(defaultHasCardinality, "Default threshold should return cardinality");
+        assertTrue(maxHasCardinality, "Max threshold should return cardinality");
+    }
+
+    @Test
+    public void facetMetricsWithSemanticQuery() {
+        SearchOptions searchOptions = new SearchOptions()
+            .setFacets("Rating, metric: min", "Rating, metric: max", "Category, metric: cardinality")
+            .setQueryType(QueryType.SEMANTIC)
+            .setSemanticSearchOptions(new SemanticSearchOptions().setSemanticConfigurationName("semantic-config"));
+        SearchPagedIterable results
+            = getSearchClientBuilder(HOTEL_INDEX_NAME, true).buildClient().search("*", searchOptions, null);
+        Map<String, List<FacetResult>> facets = results.getFacets();
+
+        assertNotNull(facets, "Facets should not be null");
+        assertTrue(facets.containsKey("Rating"), "Rating facet should be present");
+        assertTrue(facets.containsKey("Category"), "Category facet should be present");
+
+        boolean hasRatingMetrics
+            = facets.get("Rating").stream().anyMatch(facet -> facet.getMin() != null || facet.getMax() != null);
+        boolean hasCategoryMetrics = facets.get("Category").stream().anyMatch(facet -> facet.getCardinality() != null);
+
+        assertTrue(hasRatingMetrics, "Rating metrics should work with semantic query");
+        assertTrue(hasCategoryMetrics, "Category metrics should work with semantic query");
+    }
+
+    @Test
+    public void facetMetricsApiVersionCompatibility() {
+        SearchClient prevVersionClient = new SearchClientBuilder().endpoint(SEARCH_ENDPOINT)
+            .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
+            .credential(TestHelpers.getTestTokenCredential())
+            .retryPolicy(SERVICE_THROTTLE_SAFE_RETRY_POLICY)
+            .indexName(HOTEL_INDEX_NAME)
+            .serviceVersion(SearchServiceVersion.V2025_09_01)
+            .buildClient();
```

Comment:
```suggestion
        SearchClient prevVersionClient = getSearchClientBuilder(HOTEL_INDEX_NAME, true)
            .serviceVersion(SearchServiceVersion.V2025_09_01)
            .buildClient();
```

This will need to be recorded again.